pythnet/remote-executor/programs/remote-executor/src/lib.rs

```diff
@@ -1,15 +1,61 @@
-use anchor_lang::prelude::*;
+#![deny(warnings)]
+
+use anchor_lang::{prelude::*, solana_program::borsh::get_packed_len};
+use state::{claim_record::ClaimRecord, posted_vaa::PostedVaa};
+
+mod state;
+mod error;
 
 declare_id!("Fg6PaFpoGXkYsidMpWTK6W2BeZ7FEfcYkg476zPFsLnS");
 
 #[program]
 pub mod remote_executor {
+    use anchor_lang::solana_program::{program::invoke_signed, instruction::Instruction};
+
+    use crate::{state::{posted_vaa::CHAIN_ID_SOLANA, governance_payload::ExecutorPayload}, error::ExecutorError};
+
     use super::*;
 
     pub fn execute_posted_vaa(ctx: Context<ExecutePostedVaa>) -> Result<()> {
+        let posted_vaa = &ctx.accounts.posted_vaa;
+        let claim_record = &mut ctx.accounts.claim_record;
+        assert_or_err(posted_vaa.emitter_chain == CHAIN_ID_SOLANA, err!(ExecutorError::EmitterChainNotSolana))?;
+        assert_or_err(posted_vaa.sequence > claim_record.sequence, err!(ExecutorError::NonIncreasingSequence))?;
+        claim_record.sequence = posted_vaa.sequence;
+
+        let payload = ExecutorPayload::try_from_slice(&posted_vaa.payload)?;
```

Comment:
We usually parse these stuff manually. Borsh definition of the structs and their order might be different than our wire format.

Reply:
Got it. Indeed borsh was not doing it right. But I fixed it and I'll make sure it works with unit tests.